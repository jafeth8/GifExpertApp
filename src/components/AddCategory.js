import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({setcategories}) => {
    const [inputValue, setinputValue] = useState('')

    const handleInputChange=(e)=> {
        //console.log(e.target.value);
        setinputValue(e.target.value);
        //console.log("handleInputChange() ejecutado")//temporal para pruebas
    }

    const handleSubmit=(e)=> {
        e.preventDefault();
        console.log('handle event ejecutado');
        
        if(inputValue.trim().length>2){
            setcategories(cats=>[inputValue,...cats]);
            setinputValue('');
        }
        
    }
    return (
        <form onSubmit={handleSubmit}>
            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}
            />  
        </form>
    )
}

AddCategory.propTypes = {
    setcategories:PropTypes.func.isRequired,
}
