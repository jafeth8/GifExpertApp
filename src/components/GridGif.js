import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'

import { GifGridItem } from './GifGridItem';

export const GridGif = ({category}) => {

    
    const {data:images,loading}=useFetchGifs(category)
    console.log("cargando componente");
     
    return (
        <>
            <h3 className="animate__animated animate__fadeIn">{category}</h3>
            {loading?<p className="animate__animated animate__flash" >loading..</p>:null}
            <div className="card-grid">
            
                {
                    images.map(img =>(
                        <GifGridItem 
                        key={img.id} 
                        //img={img}
                        {...img}
                        />
                    ))
                }
            
            </div>

        </>
        
    )
}
