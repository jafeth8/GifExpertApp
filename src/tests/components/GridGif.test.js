import "@testing-library/jest-dom"
import {shallow} from 'enzyme'
import { GridGif } from "../../components/GridGif";

import { useFetchGifs } from "../../hooks/useFetchGifs";

jest.mock('../../hooks/useFetchGifs')


const category ="kira"

describe('pruebas en <GridGif/>', () => {

    test('debe mostrar el componente correctamente', () => {
        useFetchGifs.mockReturnValue({
            data:[],
            loading:true
        })

        const wrapper=shallow(<GridGif category={category}/>);
        expect(wrapper).toMatchSnapshot();
    })

    test('debe de cargarse los <GifGridItem/> cuando se cargan imagenes useFetchGifs', () => {
        
        const gifs=[
            {id:"Abc",title:"Cualquiera",url:"https://localhost/cualquiera.jpg"},
            {id:"Abcd",title:"Cualquiera",url:"https://localhost/cualquiera.jpg"}
        ]

        useFetchGifs.mockReturnValue({   
            data:gifs,
            loading:false
        })
        const wrapper=shallow(<GridGif category={category}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    })
    
    
})

