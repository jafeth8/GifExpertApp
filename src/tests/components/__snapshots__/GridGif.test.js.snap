// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GridGif/> debe de cargarse los <GifGridItem/> cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    kira
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="Abc"
      key="Abc"
      title="Cualquiera"
      url="https://localhost/cualquiera.jpg"
    />
    <GifGridItem
      id="Abcd"
      key="Abcd"
      title="Cualquiera"
      url="https://localhost/cualquiera.jpg"
    />
  </div>
</Fragment>
`;

exports[`pruebas en <GridGif/> debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    kira
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    loading..
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
